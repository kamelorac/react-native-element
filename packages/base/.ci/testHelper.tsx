import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
