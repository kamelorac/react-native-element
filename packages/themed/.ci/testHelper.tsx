import React, { JSXElementConstructor } from 'react';
import { View } from 'react-native';
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from '@testing-library/react-native';
import {
  ThemeProvider,
  FullTheme,
  createTheme,
  CreateThemeOptions,
} from '../src/config';

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  themeProp: Partial<CreateThemeOptions> = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };
  const renderApi = render(
    <ThemeProvider theme={createTheme(themeProp)}>{children}</ThemeProvider>,
    options
  );
  const wrapper = renderApi.queryByTestId(wrapperTestID || 'wrapper')!;
  return { wrapper, ...renderApi };
};
