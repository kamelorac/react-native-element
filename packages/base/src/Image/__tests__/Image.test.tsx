import React from 'react';
import { Image as RNImage, Text } from 'react-native';
import { Image } from '..';
import { renderWithWrapper } from '../../../.ci/testHelper';

describe('Image Component', () => {
  const FAKE_URI = 'https://i.imgur.com/0y8Ftya.jpg';
  it('should render', () => {
    const { toJSON } = renderWithWrapper(
      <Image testID="customTestId" source={{ uri: FAKE_URI }} />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the appropriate testId when one is passed.', () => {
    const { queryByTestId } = renderWithWrapper(
      <Image testID="customTestId" source={{ uri: FAKE_URI }} />
    );
    const image = queryByTestId('customTestId');
    expect(image.props.source.uri).toBe(FAKE_URI);
  });

  it('should render the appropriate testId when one is not passed.', () => {
    const { queryByTestId } = renderWithWrapper(
      <Image source={{ uri: FAKE_URI }} />
    );
    const image = queryByTestId('RNE__Image');
    expect(image.props.source.uri).toBe(FAKE_URI);
  });

  it('should render without the transition', () => {
    const { queryByTestId } = renderWithWrapper(
      <Image source={{ uri: FAKE_URI }} transition={false} />
    );
    const image = queryByTestId('RNE__Image');
    expect(image.props.onLoad).not.toBeNull();
  });

  it('should render PlaceholderContent', () => {
    const { queryByTestId } = renderWithWrapper(
      <Image
        source={{ uri: FAKE_URI }}
        PlaceholderContent={<Text testID="text">MyText</Text>}
      />
    );
    const imageText = queryByTestId('text');
    expect(imageText.props.children).toBe('MyText');
  });

  it('should apply value from style prop', () => {
    const { queryByTestId } = renderWithWrapper(
      <Image source={{ uri: FAKE_URI }} style={{ tintColor: 'red' }} />
    );
    const image = queryByTestId('RNE__Image');
    expect(image.props.style.tintColor).toBe('red');
  });

  it('should apply value from childrenContainerStyle prop', () => {
    const { queryByTestId } = renderWithWrapper(
      <Image
        source={{ uri: FAKE_URI }}
        childrenContainerStyle={{
          borderWidth: 1,
          borderColor: 'red',
        }}
      />
    );
    const childContainer = queryByTestId('RNE__Image__children__container');

    expect(childContainer.props.style).toMatchObject({
      borderWidth: 1,
      borderColor: 'red',
    });
  });

  it('static methods should be present', () => {
    expect(Image.getSize).toBe(RNImage.getSize);
    expect(Image.getSizeWithHeaders).toBe(RNImage.getSizeWithHeaders);
    expect(Image.prefetch).toBe(RNImage.prefetch);
    expect(Image.abortPrefetch).toBe(RNImage.abortPrefetch);
    expect(Image.queryCache).toBe(RNImage.queryCache);
    expect(Image.resolveAssetSource).toBe(RNImage.resolveAssetSource);
  });
});
