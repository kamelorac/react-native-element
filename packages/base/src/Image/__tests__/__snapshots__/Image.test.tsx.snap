// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component should render 1`] = `
<View
  testID="wrapper"
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "backgroundColor": "transparent",
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <Image
      onLoad={[Function]}
      source={
        {
          "uri": "https://i.imgur.com/0y8Ftya.jpg",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      testID="customTestId"
      transitionDuration={360}
    />
    <View
      accessibilityElementsHidden={true}
      collapsable={false}
      importantForAccessibility="no-hide-descendants"
      pointerEvents="none"
      style={
        {
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#bdbdbd",
            "justifyContent": "center",
          }
        }
        testID="RNE__Image__placeholder"
      />
    </View>
    <View
      style={{}}
      testID="RNE__Image__children__container"
    />
  </View>
</View>
`;
